apps/betterangels-backend/clients/types.py

```diff
@@ -104,10 +111,34 @@ def search(
 
         queryset = queryset.filter(reduce(and_, q_objects))
 
-        return (
-            queryset,
-            Q(),
-        )
+        return (queryset, Q())
+
+    @strawberry_django.filter_field
+    def full_name_search(
+        self,
+        queryset: QuerySet,
+        info: Info,
+        value: UserSearchInput,
+        prefix: str,
+    ) -> Tuple[QuerySet[ClientProfile], Q]:
+        """
+        Returns only exact match on first and last name (case insensitive).
+        Accepts optional middle name param. If middle name is passed, middle name must be an exact match as well.
+        """
+        if not value.first_name or not value.last_name:
+            return (queryset.none(), Q())
```

Comment:
@tglaz i was thinking about this endpoint specifically for deduping purposes, so that's why it requires first and last name... that might not be expected behavior so that's why i'm calling it out
like if you want to find all the people with "Anthony" in their name, you would use the traditional `search` filter above. however, that would return people with "Anthony" in either their first OR last name.